app/SwimChrono/app/src/main/java/es/udc/apm/swimchrono/services/ApiService.kt

```diff
@@ -100,9 +102,175 @@ class ApiService : Service() {
+                        snapshot.children.forEach { clubSnapshot ->
+                            val genericType = object :
+                                GenericTypeIndicator<List<HashMap<String, String>>>() {}
+
+                            clubSnapshot.child("members").getValue(genericType)?.forEach { user ->
+                                if (user.containsValue(uid)) {
+                                    val clubId = clubSnapshot.child("id").getValue(Int::class.java)
+                                    Logger.debug(tag,
+                                        "Encontrado el usuario ".plus(uid).plus(" en el club ").plus(clubId).plus(" como deportista"))
+                                    getClub(clubId, callback)
+                                    return
+                                }
+                            }
+
+                            clubSnapshot.child("trainers").getValue(genericType)?.forEach { trainer ->
+                                if (trainer.containsValue(uid)) {
+                                    val clubId = clubSnapshot.child("id").getValue(Int::class.java)
+                                    Logger.debug(tag,
+                                        "Encontrado el usuario ".plus(uid).plus(" en el club ").plus(clubId).plus(" como entrenador"))
+                                    getClub(clubId, callback)
+                                    return
+                                }
+                            }
+                        }
+
+                        Logger.error(tag, "El usuario".plus(uid).plus("no pertenece a ningún club"))
+                    }
+
+                    override fun onCancelled(error: DatabaseError) {
+                        Logger.error(tag, "Error al obtener datos de Firebase: ${error.message}")
+                    }
+                })
+            } catch (e: Exception) {
+                Logger.error(tag, "Error $e")
+            }
+        }
+    }
+
+    fun getUsers(id: Int?, userType: String, callback: ApiServiceCallback) {
+        CoroutineScope(Dispatchers.IO).launch {
+            try {
+                val clubsRef = database.child("clubs")
+                val trainersRef = database.child("users")
+                val users = mutableListOf<Any?>()
+                val genericType = object : GenericTypeIndicator<List<HashMap<String, String>>>() {}
+
+                clubsRef.addListenerForSingleValueEvent(object : ValueEventListener {
+                    override fun onDataChange(snapshot: DataSnapshot) {
+                        snapshot.children.forEach { clubSnapshot ->
+                            // Obtener el ID del club
+                            val clubId = clubSnapshot.child("id").getValue(Int::class.java)
+                            // Verificar si el id del club coincide con el club buscado
+                            if (clubId == id) {
+                                if (userType == "trainers") {
```

Comment:
He intentado definir una cláusula if en función del valor del rol del usuario `userType` para extraer los entrenadores o los miembros de un club (`clubSnapshot.child("trainers").getValue(genericType)` y `clubSnapshot.child("members").getValue(genericType)`, respectivamente).

Con todo, no he sido capaz. Esto supone la duplicación del código para obtener los datos de los usuarios. Si sabéis de alguna solución, por favor, responded a este hilo y lo implementaré.